builtin/credential/ldap/backend.go

```diff
@@ -155,23 +180,33 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 		return nil, logical.ErrorResponse(fmt.Sprintf("LDAP search failed: %v", err)), nil
 	}
 
-	var allgroups []string
-	var policies []string
+	for _, e := range sresult.Entries {
+		dn, err := ldap.ParseDN(e.DN)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, rdn := range dn.RDNs {
+			for _, rdnTypeAndValue := range rdn.Attributes {
```

Comment:
I get what you're doing here...my question is, are you sure that "CN" is the type for group values across all LDAP servers?